### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by 0bi0n3**
Attempted steps 1-12, encountered some errors when compiling but attempted fixes prior to sending to review. Code compiles, I am unsure if results are correct.

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -56,48 +56,63 @@ Purpose:  This project will show you the difference between member functions and
...
-struct <#structName1#>                                //4
+struct CompareFunction                               //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if( a->a < b->a ) return a;
+        if( a->a > b->a ) return b;
```
Comment (matkatmusic) [inline main.cpp]: The member variable was called `value`.  You did not need to rename it
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -56,48 +56,63 @@ Purpose:  This project will show you the difference between member functions and
...
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float var1 { 0.0 }, var2 { 0.0 };
+    float multiply(float* updatedValue)      //12
     {
-        
+        std::cout << "U's var1 value: " << this->var1 << std::endl;
+
+        this->var1 = *updatedValue;
+
+        std::cout << "U's var1 updated value: " << this->var1 << std::endl;
```
Comment (matkatmusic) [inline main.cpp]: What do we always do before using a pointer?
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -56,48 +56,63 @@ Purpose:  This project will show you the difference between member functions and
...
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float multiply(U* that, float* newValue)        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        std::cout << "U's var1 value: " << that->var1 << std::endl;
+        that->var1 = *newValue;
+        std::cout << "U's var1 updated value: " << that->var1 << std::endl;
```
Comment (matkatmusic) [inline main.cpp]: What do we always do before using a pointer?
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 1 above)
Comment (0bi0n3) [reply main.cpp]: I think I see it!
I didn't need to match the a & b variables, these are already provided in the `T* compare(T* a, T* b)` function.

Whereas the constructor in `T(float v, const char* str) : value(v), name(str) {}` hints at the variable name with `v`?

Please let me know if I have misunderstood that.

File: main.cpp (hunk 2 above)
Comment (0bi0n3) [reply main.cpp]: Check for `nullptr`!

Fixed this with an `if` conditional.

File: main.cpp (hunk 3 above)
Comment (0bi0n3) [reply main.cpp]: Check for `nullptr`!

Fixed this with two `if` conditionals for both pointers.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**